blockchain/types/transaction.go

```diff
@@ -198,6 +216,19 @@ func (tx *Transaction) DecodeRLP(s *rlp.Stream) error {
 	return nil
 }
 
+// UnmarshalBinary decodes the canonical encoding of transactions.
+// It supports legacy RLP transactions and EIP2718 typed transactions.
+func (tx *Transaction) UnmarshalBinary(b []byte) error {
+	newTx := &Transaction{}
+	if err := rlp.DecodeBytes(b, newTx); err != nil {
```

Comment:
Why don't you use `tx.DecodeRLP` ?

Reply:
I know that tx.DecodeRLP() is an internal function that is automatically called when rlp.decodeBytes is called, and that it is not called directly.